gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -80,14 +85,38 @@ func NewDriverRemoteConnection(
 		transporterType: settings.TransporterType,
 		logHandler:      logHandler,
 		connection:      connection,
+		session:         settings.Session,
+		closed:          false,
 	}
 
 	return &DriverRemoteConnection{client: client}, nil
 }
 
 // Close closes the DriverRemoteConnection.
 func (driver *DriverRemoteConnection) Close() error {
-	return driver.client.Close()
+	// If DriverRemoteConnection has spawnedSessions then they must be closed as well.
+	if len(driver.spawnedSessions) > 0 {
+		driver.client.logHandler.logger.Logf(Info, "closing spawned sessions from DriverRemoteConnection with url '%s'", driver.client.url)
+		for _, session := range driver.spawnedSessions {
+			err := session.Close()
+			if err != nil {
+				return err
+			}
+		}
+		driver.spawnedSessions = []*DriverRemoteConnection{}
+	}
+
+	if driver.IsSessionBound() {
+		driver.client.logHandler.logger.Logf(Info, "closing DriverRemoteConnection with url '%s' with session '%s'", driver.client.url, driver.client.session)
+	} else {
+		driver.client.logHandler.logger.Logf(Info, "closing DriverRemoteConnection with url '%s'", driver.client.url)
+	}
```

Comment:
I think you should be using our logger in the other fashion with the resources.

Reply:
Im not sure I understand what you mean

Reply:
Check out `logger.go` and the `resources` folder and see how log messages are structured (note that even though the type is called `errorKey`, we are logging state, it's just a bad name for now)